scope_test.go

```diff
@@ -529,6 +575,36 @@ func TestTaggedSubScope(t *testing.T) {
 	}, r.histograms["foo.bar"].tags)
 }
 
+func TestTaggedSanitisedSubScope(t *testing.T) {
+	r := newTestStatsReporter()
+
+	ts := map[string]string{"env": "test:env"}
+	root, closer := NewRootScope(ScopeOptions{
+		Prefix:          "foo",
+		Tags:            ts,
+		Reporter:        r,
+		SanitiseOptions: &alphanumericSanitiserOpts,
+	}, 0)
+	defer closer.Close()
+
+	s := root.(*scope)
+
+	tscope := root.Tagged(map[string]string{"service": "test.service"}).(*scope)
+
+	r.cg.Add(1)
+	tscope.Counter("beep").Inc(1)
+
+	s.report(r)
+	tscope.report(r)
+	r.cg.Wait()
+
+	assert.EqualValues(t, 1, r.counters["foo.beep"].val)
```

Comment:
Sanitizing the output of fullyQualifiedName seems more like what the user is asking for if they pass you a sanitizer, even if it's not "as powerful" per se.  i.e. no way to get around it if we do that.  

For instance if we do it on the output of fullyQualifiedName then for scopes using the Prometheus reporter we wouldn't have to set the default tag separator to something different, we would just have to set the sanitizer to UnderscoreDashSeparated and we'd be done - we get sanitization for our input and for the prefixing too.